emperor/flat_plots.py

```diff
@@ -0,0 +1,352 @@
+        identifiers in the ``ordination`` object.
+    field : str
+        The field within the mapping file to use for coloring.
+    colors : dict or str
+        If a dict, it is expected to provide a mapping of category value to
+        a color; entries not in the mapping will be considered "null". If a
+        str, the value is interpreted as the name of a color map and assumes
+        a gradient is to be painted.
+    null_kw : dict, optional
+        Parameters to use for plotting and painting null or unknown values.
+    focus_kw : dict, optional
+        Parameters to use for plotting and painting values which are not
+        null and in the field of interest.
+    """
+    # setup defaults if not provided
+    if null_kw is None:
+        null_kw = {}
+    if focus_kw is None:
+        focus_kw = {}
+
+    # resolve if this is discrete or continous
+    if isinstance(colors, str):
+        plot = two_dimensional_gradient
+        legend = legend_gradient
+    else:
+        plot = two_dimensional_discrete
+        legend = legend_discrete
+
+    # full page, with square plots
+    fig, grid = plt.subplots(3, 4, sharex=False, sharey=False,
+                             figsize=(24, 16))
+    grid[0, 3].grid(False)
+    grid[0, 3].set_axis_off()
+    grid[2, 3].grid(False)
+    grid[2, 3].set_axis_off()
+
+    for row in range(3):
+        # y_pc is the axis in the ordination to put on the y-axis
+        y_pc = row + offset
+
+        # the y-axis is consistent across a row so only label the
+        # first column
+        row_ax = grid[row, 0]
+        row_ax.set_ylabel(_pcoa_label(y_pc, ordination), fontsize=14)
+
+        for col in range(3):
+            # x_pc is the axis in the ordination to put on the x-axis
+            x_pc = row + col + offset + 1
+
+            # fetch the specific subplot, and plot
+            ax = grid[row, col]
+            ax.set_xlabel(_pcoa_label(x_pc, ordination), fontsize=14)
+            plot(ax, field, mapping, ordination, x_pc, y_pc, colors, null_kw,
+                 focus_kw)
+
+    # establish the legend on the right side
+    legend(grid[1, 3], field, mapping, ordination, colors, null_kw)
+
+    return (fig, grid)
+
+
+def _pcoa_label(dim, ordination):
+    """Helper method to make a PCoA label"""
+    fmt = 'PC%d (%0.2f%%)'
+    return fmt % (dim + 1, ordination.proportion_explained[dim] * 100)
+
+
+def two_dimensional_gradient(ax, field, mapping, ordination, xaxis, yaxis,
+                             colormap, null_kw, focus_kw):
+    """Plot two axes
+
+    Parameters
+    ----------
+    ax : AxesSubplot
+        A subplot to plot into
+    field : str
+        A metadata field in the mapping to plot.
+    mapping : pd.DataFrame
+        The metadata about the samples.
+    ordination : skbio.OrdinationResults
+        The ordination object.
+    xaxis : integer
+        An axis to plot from the ordination.
```

Comment:
hmm - that's missing quite a few use cases.

Most of the scikit-bio instances of the OrdinationResults objects are populated with string labels

https://github.com/biocore/scikit-bio/blob/master/skbio/stats/ordination/_principal_coordinate_analysis.py#L136